### PR TITLE
Implement RH repository entity + some fixes for 6.5

### PR DESCRIPTION
**Pull request by ldjebran**
Implement RH repository entity and fix repository entity.
test coverage at https://github.com/SatelliteQE/robottelo/pull/6448

### CODE REVIEW

File: airgun/views/redhat_repository.py [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+    TextInput,
+    View,
+)
+
+from airgun.views.common import BaseLoggedInView
+from airgun.widgets import (
+    ActionsDropdown,
+)
+
+SPINNER_LOCATOR = ".//div[contains(@class, 'spinner')]"
```
Comment (abalakh) [inline airgun/views/redhat_repository.py]: maybe move it into _wait_for_spinner() as it's never used elsewhere?

File: airgun/views/redhat_repository.py [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+            if item == arch_version_item.text:
+                arch_version_item.enable()
+                break
+        else:
+            raise ValueError(
+                'Repository "{0}" was not found in repository set "{1}"'.format(item, self.name))
+
+    def fill(self, value):
+        """Fill this widget with value, this will enable the repository name in this repository set
+            if value is True."""
```
Comment (abalakh) [inline airgun/views/redhat_repository.py]: nitpick: redundant indentation

File: airgun/views/redhat_repository.py [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+        return self.text
+
+    def enable(self):
+        """Enable this repository."""
+        self.browser.click(self.ENABLE_BUTTON, parent=self)
+        _wait_for_spinner(self.parent)
+
+    def fill(self, value):
+        """Fill this widget by enabling the repository if value is True"""
+        if value is True:
```
Comment (abalakh) [inline airgun/views/redhat_repository.py]: maybe we should rise some exception if value is smth else?

File: airgun/views/redhat_repository.py [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+    DISABLE_BUTTON = ".//button"
+
+    def disable(self):
+        """Disable this repository."""
+        self.browser.click(self.DISABLE_BUTTON, parent=self)
+        _wait_for_spinner(self.parent)
+
+    def fill(self, value):
+        """Fill this widget with value, this will disable this repository if value is False."""
+        if value is False:
```
Comment (abalakh) [inline airgun/views/redhat_repository.py]: >maybe we should rise some exception if value is smth else?

File: airgun/views/repository.py [hunk 5]
```diff
@@ -122,6 +122,28 @@ def is_displayed(self):
...
+    clear_button = Text(
+        ".//span[contains(@ng-hide, 'editMode')]/i[@ng-show='deletable']")
+
+    @View.nested
+    class edit_field(View):
+        username = TextInput(id='upstream_username')
+        password = TextInput(id='upstream_password')
+
+    def fill(self, value):
+        """Handle the clear functionality, if username is supplied with None or empty
```
Comment (abalakh) [inline airgun/views/repository.py]: Wouldn't it be better to update EditableEntry's fill() instead? (if None or empty string - hit 'clear' button)?

File: airgun/views/common.py [hunk 6]
```diff
@@ -28,7 +28,8 @@
 
 
 class BaseLoggedInView(View):
-    menu = SatVerticalNavigation('.//div[@id="vertical-nav"]/ul')
+    menu = SatVerticalNavigation(
+        './/div[@id="vertical-nav" or contains(@class, "nav-pf-vertical")]/ul')
```
[the remark this answers is not in the document]
Comment (abalakh) [reply airgun/views/common.py]: Pls disregard as per #235

**Comment by ldjebran** [comment]
rebased

File: airgun/views/redhat_repository.py (hunk 3 above)
Comment (ldjebran) [reply airgun/views/redhat_repository.py]: I think we should not be explicit here. if True enable , if not ignore, that's what I think, this does not worth additional work from the user.

File: airgun/views/redhat_repository.py (hunk 4 above)
Comment (ldjebran) [reply airgun/views/redhat_repository.py]: I think we should not be explicit here. if False disable , if not ignore, that's what I think, this does not worth additional work from the user.

File: airgun/session.py [hunk 7]
```diff
@@ -445,6 +446,10 @@ def provisioningtemplate(self):
         """Instance of Provisioning Template entity."""
         return ProvisioningTemplateEntity(self.browser)
 
+    @cached_property
+    def redhat_repository(self):
```
[the remark this answers is not in the document]
Comment (ldjebran) [reply airgun/session.py]: Think this a simple python property here, do not really understand why we are implementing other naming convention  than python recommend.

[the remark this answers is not in the document]
Comment (oshtaier) [reply airgun/session.py]: yeah `redhatrepository`. It is just a style used across all entities

**Comment by ldjebran** [comment]
@abalakh @oshtaier most of the comments addressed, test output https://github.com/SatelliteQE/robottelo/pull/6448#issuecomment-438196155

File: airgun/entities/redhat_repository.py [hunk 8]
```diff
@@ -0,0 +1,66 @@
+from airgun.entities.base import BaseEntity
+from airgun.navigation import NavigateStep, navigator
+
+from airgun.views.redhat_repository import RedHatRepositoriesView
+
+
+class RedHatRepositoryEntity(BaseEntity):
+
+    def search(self, value, category='Available', types=None):
```
[the remark this answers is not in the document]
Comment (abalakh) [reply airgun/entities/redhat_repository.py]: @oshtaier your vote? :)

[the remark this answers is not in the document]
Comment (oshtaier) [reply airgun/entities/redhat_repository.py]: doesn't seem so important

**Comment by ldjebran** [comment]
removed the return from delete operation

**Comment by ldjebran** [comment]
test output https://github.com/SatelliteQE/robottelo/pull/6448#issuecomment-438253343

File: airgun/session.py [hunk 9]
```diff
@@ -445,6 +446,10 @@ def provisioningtemplate(self):
         """Instance of Provisioning Template entity."""
         return ProvisioningTemplateEntity(self.browser)
 
+    @cached_property
+    def redhatrepository(self):
```
Comment (oshtaier) [inline airgun/session.py]: forgot a docstring

**Comment by ldjebran** [comment]
@oshtaier updated

File: airgun/views/redhat_repository.py [hunk 10]
```diff
@@ -0,0 +1,269 @@
...
+        """Return all the items (available repositories) of this repository set."""
+        if self.ITEM:
+            self.expand()
+            return [
+                AvailableRepositoryItem(self, '{0}[{1}]'.format(self.ITEMS, index + 1))
+                for index, _ in enumerate(self.browser.elements(self.ITEMS, parent=self))
+            ]
+        return []
+
+    def read_items(self):
```
Comment (oshtaier) [inline airgun/views/redhat_repository.py]: why do we need both read and read_items? Why don't we combine them into one?

File: airgun/views/redhat_repository.py (hunk 4 above)
Comment (oshtaier) [reply airgun/views/redhat_repository.py]: not sure that I get the logic, it is still like:
```
def fill(self, value):
    if value is True:
        self.disable()
```

File: airgun/views/redhat_repository.py (hunk 10 above)
Comment (ldjebran) [reply airgun/views/redhat_repository.py]: The read will read only the name and label to read the repo set items we have to expand the repository set that will make a server call to get the items, it's same way that when we have a table we read only the table items of the the entity but to read the entity we have to open it and read the edit form , I'am using the same analogy as for simple entity, as if we have 20 entities in the table we are not going to open each entity and read it.

Comment (ldjebran) [reply airgun/views/redhat_repository.py]: EG: we need this only when we are reading a concrete defined entity.

File: airgun/views/redhat_repository.py (hunk 4 above)
Comment (ldjebran) [reply airgun/views/redhat_repository.py]: @oshtaier Sorry I did not get your point here.

**Comment by ldjebran** [comment]
@oshtaier doc of entity read corrected the fill functions removed
```
(robottelo) dlezz@dlezz:~/projects/robottelo-fork$ pytest tests/foreman/ui_airgun/test_repository.py -v -k "test_positive_reposet_disable or test_positive_reposet_disable_after_manifest_deleted"
============================================ test session starts =============================================
platform linux -- Python 3.6.6, pytest-3.6.1, py-1.5.4, pluggy-0.6.0 -- /home/dlezz/.pyenv/versions/3.6.6/envs/robottelo/bin/python
cachedir: .pytest_cache
shared_function enabled - ON - scope: ffff - storage: file
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: wait-for-1.0.9, xdist-1.22.5, services-1.3.0, mock-1.10.0, forked-0.2, cov-2.5.1
collecting 17 items                                                                                          2018-11-14 14:43:38 - conftest - DEBUG - BZ deselect is disabled in settings

collected 17 items / 15 deselected                                                                           

tests/foreman/ui_airgun/test_repository.py::test_positive_reposet_disable PASSED                       [ 50%]
tests/foreman/ui_airgun/test_repository.py::test_positive_reposet_disable_after_manifest_deleted PASSED [100%]

================================= 2 passed, 15 deselected in 395.21 seconds ==================================
```

**Pull request merged**